lib/db/open.go

```diff
@@ -0,0 +1,51 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"math/rand"
+	"time"
+
+	_ "github.com/lib/pq"
+)
+
+type dbOpts struct {
+	maxConnectionLifetime       time.Duration
+	maxConnectionLifetimeJitter time.Duration
+}
+
+type PGOpenOption func(*dbOpts)
+
+func WithMaxConnectionLifetime(d time.Duration) PGOpenOption {
+	return func(opts *dbOpts) {
+		opts.maxConnectionLifetime = d
+	}
+}
+
+func WithMaxConnectionLifetimeJitter(d time.Duration) PGOpenOption {
+	return func(opts *dbOpts) {
+		opts.maxConnectionLifetimeJitter = d
+	}
+}
+
+func PGOpen(dataSourceName string) (*sql.DB, error) {
+	return PGOpenContext(context.Background(), dataSourceName)
+}
+
+func PGOpenContext(ctx context.Context, dataSourceName string) (*sql.DB, error) {
+	opts := dbOpts{
+		maxConnectionLifetime:       10 * time.Minute,
+		maxConnectionLifetimeJitter: 10 * time.Minute,
```

Comment:
Is the idle connection timeout for the HA cluster 5 or 6 minutes? Should this total to less than that?

Reply:
oh, yea
forgot about that

Reply:
maybe we do 5 - jitter(1min)?

Reply:
It looks like the timeout in HA cluster might be 50s:

https://github.com/chef/a2-ha-backend/blob/master/components/automate-backend-haproxy/config/haproxy.conf#L10-L13